odoo/models.py

```diff
@@ -2413,6 +2419,19 @@ def _read_group_format_result(self, data, annotated_groupbys, groupby, domain):
         del data['id']
         return data
 
+    @api.model
+    def _read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        """
+        Executes exactly what the public read_group, except it doesn't order the manyToOne on their names but on their ID instead
+        """
+        if not orderby:
+            if isinstance(groupby, str):
+                groupby = [groupby]
+            groupby_list = groupby[:1] if lazy else groupby
+            groupby_list = [fname + '.id' if self._fields[fname].type == 'many2one' else fname for fname in groupby_list]
```

Comment:
This is gonna crash, as `fname` is not necessarily a field name.  Think about `'date:month'`...